rs/sns/governance/src/governance.rs

```diff
@@ -2283,6 +2284,65 @@ impl Governance {
         }
     }
 
+    async fn perform_register_extension(
+        &mut self,
+        register_extension: RegisterExtension,
+    ) -> Result<(), GovernanceError> {
+        // Step 0. Validate the RegisterExtension proposal.
+        let ValidatedRegisterExtension { wasm, init, spec } =
+            register_extension.try_into().map_err(|err| {
+                GovernanceError::new_with_message(
+                    ErrorType::InvalidProposal,
+                    format!("Invalid RegisterExtension: {err:?}"),
+                )
+            })?;
+
+        let Wasm::Chunked {
+            wasm_module_hash,
+            store_canister_id,
+            chunk_hashes_list,
+        } = wasm
+        else {
+            return Err(GovernanceError::new_with_message(
+                ErrorType::InvalidProposal,
+                "RegisterExtension proposal must contain a chunked wasm module.",
+            ));
+        };
+
+        // Step 1. Register the extension as a dapp canister.
+        self.perform_register_dapp_canisters(RegisterDappCanisters {
```

Comment:
I assume we are going to switch this to be register_extension?   Root should have a clue which is which for the API return, IMO

Reply:
We discussed this before, and yes, I agree that Root would need to be able to distinguish regular dapp canisters and extensions. This could be done by adding a new parameter to `perform_register_dapp_canisters` or by adding a new Root API function (possibly, a cleaner approach). It's on my list.